### PR TITLE
fix: adm update metrics to be consistent

### PR DESCRIPTION
**Pull request by tiftran**
Currently, we only have metrics for updates for changes that occur for the adm filters. This made it hard to determine whether there was an attempt at an update or whether there were no changes. Added metrics for check attempts.

In addition, changed metric from adm.filter to filter.adm for consistency

### CODE REVIEW

**Comment by tiftran** [comment]
i feel like there should probably be tests for this? not sure how to write them though, couldn't find anything existing i could work off from either...

File: src/adm/filter.rs [hunk 1]
```diff
@@ -121,12 +121,13 @@ pub fn spawn_updater(
         let mut tags = crate::tags::Tags::default();
         loop {
             {
+                metrics.incr("filter.adm.update_check").ok();
```
Comment (ncloudioj) [inline src/adm/filter.rs:124]: I'd recommend injecting metrics into every arm of this match expression to get the whole picture for each update attempt.

```rs
    {
        match mfilter.read().await.requires_update(&storage_client).await {
            Ok(true) => {
                let mut filter = mfilter.write().await;
                match filter.update(&storage_client).await {
                    Ok(_) => {
                        metrics.incr("adm.filter.update.ok").ok();
                    }
                    Err(e) => {
                        filter.report(&e, &mut tags);
                        metrics.incr("adm.filter.update.error").ok();
                    }
                }
                // filter.update(&storage_client).await.unwrap_or_else(|e| {
                // metrics.incr("adm.filter.update.error").ok();
                // filter.report(&e, &mut tags);
                // });
            }
            Ok(false) => {
                metrics.incr("adm.filter.update.check.skip").ok();
            }
            Err(e) => {
                mfilter.read().await.report(&e, &mut tags);
                metrics.incr("adm.filter.update.check.error").ok();
            }
        }
    }
```
What do you think?

**Comment by ncloudioj** [comment]
> i feel like there should probably be tests for this? not sure how to write them though, couldn't find anything existing i could work off from either...

Looks like we don't have any tests for metrics in the unit/integration tests. So your PR is fine :)

FWIW, Merino-rs uses a logging sink for metrics in tests, perhaps we can do the same for Contile in the future.

File: src/adm/filter.rs (hunk 1 above)
Comment (tiftran) [reply src/adm/filter.rs:124]: thorough sounds good

**Comment by pjenvey** [comment]
Contile does have a couple uses of the `SpyMetricsSink` you could test this with, see integration tests (`init_app_with_spy` in `web/test.rs`) and the `img_storage` module also has another example.

**Review by ncloudioj — APPROVED** [review]
Looks good. Looks like CI busted. We might need to add `RUSTSEC-2021-0141` to https://github.com/mozilla-services/contile/blob/main/.cargo/audit.toml. Longer term, let's see if we can drop `dotenv`.

@pjenvey pointed out that `SpyMetricsSink` was available for testing, can you try leveraging it for some test coverage?

**Comment by tiftran** [comment]
created a separate pr for ignoring dotenv rustsec https://github.com/mozilla-services/contile/pull/497

**Review by ncloudioj — APPROVED** [review]

**Comment by tiftran** [comment]
eek had import troubles cause this dummy didn't place the test inside the test module. @pjenvey thank you for helping silly me

**Review by pjenvey — APPROVED** [review]

**Pull request merged**